src/utils/hooks/useDataSourceForVolumeSnapshot.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  DataImportCronModelGroupVersionKind,
+  DataSourceModelGroupVersionKind,
+} from '@kubevirt-ui/kubevirt-api/console';
+import {
+  V1beta1DataImportCron,
+  V1beta1DataSource,
+} from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { VolumeSnapshotKind } from '@kubevirt-utils/components/SelectSnapshot/types';
+import { getLabel, getNamespace } from '@kubevirt-utils/resources/shared';
+import { DATA_SOURCE_CRONJOB_LABEL } from '@kubevirt-utils/resources/template';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseDataSourceForVolumeSnapshot = (
+  volumeSnapshot: VolumeSnapshotKind,
+) => null | V1beta1DataSource;
+
+const useDataSourceForVolumeSnapshot: UseDataSourceForVolumeSnapshot = (volumeSnapshot) => {
+  const [dataImportCron] = useK8sWatchResource<V1beta1DataImportCron>({
+    groupVersionKind: DataImportCronModelGroupVersionKind,
+    name: getLabel(volumeSnapshot, DATA_SOURCE_CRONJOB_LABEL),
+    namespace: getNamespace(volumeSnapshot),
+  });
+
+  const [dataSource] = useK8sWatchResource<V1beta1DataSource>({
+    groupVersionKind: DataSourceModelGroupVersionKind,
+    name: dataImportCron?.spec?.managedDataSource,
+    namespace: getNamespace(dataImportCron),
```

Comment:
u are calling dataImportCron only for its ns? no other way?

Reply:
I'm using dataImportCron to get the name of the DataSource that's using the VolumeSnapshot. Is there an easier way to connect the VolumeSnapshot to the DataSource?